### DISCUSSION TITLE
How to build with external dependencies?

### DISCUSSION THREAD

**Discussion by ghost** [SpartanJ/ecode#245] [General]
I found that it will always build it own versions of libraries. In MSYS2, I already have these libraries installed. I want it to not build its own but use the MSYS2 libraries. How can I do that? Thank you.

**Comment by SpartanJ** [comment]
It's only possible for freetype2 and SDL2, the rest of the third-party dependencies need to be built statically.
Use as premake parameter `--with-dynamic-freetype` to build dynamically freetype2. SDL2 always builds dynamically. If you want to use OpenAL instead of SDL2 sound backend, use `--without-mojoal`.
This is something that can be improved (we could dynamically build: libpng, libogg, libvorbis, mbedtls, libzip, zlib), but premake files need some modifications, it's not hard, I just didn't care because adding more build options will always add more complexity and maintenance. Dynamic build makes only sense if ecode is built as a distribution package, otherwise you'll have a dependency hell.
